### ISSUE TITLE
AllureJS breaks with Cucumber 6 in parallel execution

### ISSUE THREAD

**Issue by sgkiokas** [allure-framework/allure-js#407]
**Dependencies used**
```
- "allure-cucumberjs": "^2.0.0-beta.14",
- "cucumber": "^6.0.5",
```

**The problem**
I am using Cucumber for my Test suite and I want to use Allure for the report. Hence, I created a class like:

```
const { CucumberJSAllureFormatter } = require('allure-cucumberjs');
const { AllureRuntime } = require('allure-cucumberjs');
const reportResults = './report/allure-results';

function Reporter (options) {
  return new CucumberJSAllureFormatter(
    options,
    new AllureRuntime({ resultsDir: reportResults }),
    {}
  );
};

Reporter.prototype = Object.create(CucumberJSAllureFormatter.prototype);
Reporter.prototype.constructor = Reporter;

exports.default = Reporter;
```

Then, I call that like:

```
const FEATURE_FILES_DIR = 'features/';
const ALLURE_REPORTER_DIR = './allureReporter.js';

module.exports = {
 'default': `${FEATURE_FILES_DIR} -f ${ALLURE_REPORTER_DIR}`,
}
```

This was working perfectly fine with Cucumber 4, however, since I need to take advantage of the fixes introduced for the parallel test execution, I updated Cucumber to 6. However, I'm getting the following error on the `onTestStepStarted` event:

```
*tests/node_modules/allure-cucumberjs/dist/src/CucumberJSAllureReporter.js:211
        if (location.sourceLocation !== undefined && feature.stepMap !== undefined) {
```

If I inspect the `location` object, I see that this is `undefined`.

Is this a known issue? Can I try something else?

**Where it was run**
 - OS: MacOS Big Sure 11.4
 - The intention is to be used on Jenkins

**Comment by fairywang325** [comment]
the same question,do you have any solution？

**Comment by baev** [comment]
fixed in 2.0.0-beta.18 (requires cucumberjs 8)
